File: bms/go/triangolatte/Point_Distance2/Point_Distance2.go
package test

var ()

const ()

func (p Point) Distance2(r Point) float64 {
	return (p.X-r.X)*(p.X-r.X) + (p.Y-r.Y)*(p.Y-r.Y)
}

type Point struct {
	X, Y float64
}

File: translations_only/fail/go/triangolatte/Point_Distance2/claude3/BaseRepair/Point_Distance2.rs

struct Point {
    x: f64,
    y: f64,
}

fn distance2(p: Point, r: Point) -> f64 {
    ((p.x - r.x) * (p.x - r.x) + (p.y - r.y) * (p.y - r.y))
}
